assembly/pom.xml

```diff
@@ -160,6 +160,12 @@
           <artifactId>spark-connect_${scala.binary.version}</artifactId>
           <version>${project.version}</version>
         </dependency>
+        <dependency>
```

Comment:
I found that I have to add `avro` as a provided dependency in profile `connect`.
otherwise, after `build/sbt -Pconnect clean package` it can not correctly find the avro class.
cc @HyukjinKwon